File: updated_code/code/java/1903-largest-odd-number-in-string.java
package com.example.solution;

import java.util.StringJoiner;

public class Solution {
    public String largestOddNumber(String num) {
        StringJoiner result = new StringJoiner("");
        for (int i = num.length() - 1; i >= 0; i--) {
            if ((num.charAt(i) - '0') % 2 != 0) {
                for (int j = i; j >= 0; j--) {
                    if (isOddInteger(num.substring(j, i + 1))) {
                        result.add(num.substring(j, i + 1));
                    }
                }
            }
        }
        return result.isEmpty() ? "" : result.toString();
    }

    private boolean isOddInteger(String s) {
        int val = Integer.parseInt(s);
        return val % 2 != 0;
    }
}

File: updated_code/code/python/1903-largest-odd-number-in-string.py
class Solution:
    """
    This class is used to solve the problem of finding the largest odd number in a given string.
    """

    def largest_odd_number(self, num: str) -> str:
        """
        This method returns the largest-valued odd integer that is a non-empty substring of `num`, or an empty string if no odd integer exists.
        """
        for i in range(len(num) - 1, -1, -1):
            if int(num[i]) % 2 != 0:
                return num[i:]
        return ""